.github/workflows/main.yaml

```diff
@@ -45,3 +45,19 @@ jobs:
       track-name: 1.3
     secrets:
       CHARMCRAFT_AUTH: ${{ secrets.CHARMCRAFT_AUTH }}
+
+  publish-charm-on-push:
+    name: Publish Developer Charm To Branch
+    needs:
+      - lint-report
+      - static-analysis
+      - unit-tests-with-coverage
+      - integration-test
+    if: ${{ (github.ref_name != 'main') && (github.event_name == 'push') }}
```

Comment:
Do we need to remove main branch from pull_requests (line 6) ?

Reply:
No, we are not acting on pull request event here, just on push. So basically if:

* This is a push
* It is not on main

Then we will publish the charm. This allows for testing of charms even before the pull request is created